strax/context.py

```diff
@@ -626,14 +626,14 @@ def _context_hash(self):
         If any item changes in the config, so does this hash.
 
         """
-        base_hash_on_config = self.config.copy()
+        self._base_hash_on_config = self.config.copy()
```

Comment:
I think I will also save `depends_on` and `provides` to `_base_hash_on_config`, just to be safe.

Reply:
> Why _base_hash_on_config need to be an attribute of Context?
Not needed, but it was quite handy for debugging. So I think we should keep it.

> I think I will also save depends_on and provides to _base_hash_on_config, just to be safe.
This is done implicitly by looping over the plugin class registry. It will change if a plugin has additional, fewer, or different provides than the default ones.